include/exadg/functions_and_boundary_conditions/interface_coupling.h

```diff
@@ -533,175 +142,89 @@ class InterfaceCoupling
   }
 
   void
   update_data(VectorType const & dof_vector_src)
   {
-    VectorTypeDouble         dof_vector_src_double_copy;
-    VectorTypeDouble const * dof_vector_src_double_ptr;
-    if(std::is_same<double, Number>::value)
-    {
-      dof_vector_src_double_ptr = reinterpret_cast<VectorTypeDouble const *>(&dof_vector_src);
-    }
-    else
-    {
-      dof_vector_src_double_copy = dof_vector_src;
-      dof_vector_src_double_ptr  = &dof_vector_src_double_copy;
-    }
+#if DEAL_II_VERSION_GTE(10, 0, 0)
+    dof_vector_src.update_ghost_values();
+#else
+    VectorTypeDouble dof_vector_src_double;
+    dof_vector_src_double = dof_vector_src;
+    dof_vector_src_double.update_ghost_values();
+#endif
 
-    dof_vector_src_double_ptr->update_ghost_values();
-
-    /*
-     * 1. Interpolate data on src-side
-     */
     for(auto quadrature : quad_rules_dst)
     {
-      ArrayVectorTensor & array_solution_dst = map_solution_dst.find(quadrature)->second;
-
-      std::map<mpi_rank, ArrayVectorCache> & mpi_map_cache_src =
-        map_cache_src.find(quadrature)->second;
-      std::map<mpi_rank, ArrayVectorTensor> & mpi_map_solution_src =
-        map_solution_src.find(quadrature)->second;
-
-      for(auto it : mpi_map_cache_src)
-      {
-        mpi_rank const      proc               = it.first;
-        ArrayVectorCache &  array_cache_src    = mpi_map_cache_src.find(proc)->second;
-        ArrayVectorTensor & array_solution_src = mpi_map_solution_src.find(proc)->second;
-
-        for(types::global_dof_index q = 0; q < array_cache_src.size(); ++q)
-        {
-          std::vector<Cache> &                     vector_cache    = array_cache_src[q];
-          std::vector<Tensor<rank, dim, double>> & vector_solution = array_solution_src[q];
-
-          // interpolate solution from dof vector using cached data
-          for(unsigned int i = 0; i < vector_cache.size(); ++i)
-          {
-            vector_solution[i] = Interpolator<rank, dim, double>::value(*dof_handler_src,
-                                                                        *dof_vector_src_double_ptr,
-                                                                        vector_cache[i].first,
-                                                                        vector_cache[i].second);
-          }
-        }
-      }
-
-      /*
-       * 2. Communication: transfer results back to dst-side
-       */
-      InterfaceCommunicator<dim, dim> & communicator = map_communicator.find(quadrature)->second;
-
-      communicator.process(mpi_map_solution_src, array_solution_dst, 11 + quadrature);
+#if DEAL_II_VERSION_GTE(10, 0, 0)
+      auto const result =
+        VectorTools::point_values<n_components>(map_communicator[quadrature],
+                                                *dof_handler_src,
+                                                dof_vector_src,
+                                                VectorTools::EvaluationFlags::avg);
```

Comment:
@nfehn Averaging is done by deal.II so that the vectors of vectors could be removed.

Reply:
Is this comment still relevant? It seems as if you already removed the vector of vectors?

Reply:
Yes, it has been removed.